src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.targets

```diff
@@ -142,6 +142,11 @@ The .NET Foundation licenses this file to you under the MIT license.
       <StaticICULibs Include="-Wl,-Bdynamic" Condition="'$(StaticExecutable)' != 'true'" />
     </ItemGroup>
 
+    <ItemGroup Condition="'$(StaticOpenSslLinking)' != 'true' and Exists('$(IlcSdkPath)nonportable.txt')">
```

Comment:
Right, if the logic gets any more complex, it is likely to have interactions with other logic in this file and it is not clear whether it would work well. We can wait for a concrete use-case.

Reply:
> it is likely to have interactions with other logic in this file and it is not clear whether it would work well.

Yes, a much larger portion of the file would probably move into the pack.